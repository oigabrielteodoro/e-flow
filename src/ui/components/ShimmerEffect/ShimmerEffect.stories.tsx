import { Story, Meta } from '@storybook/react/types-6-0'

import { ShimmerEffect } from '.'

export default {
  title: 'ShimmerEffect',
  component: ShimmerEffect,
} as Meta

export const Basic: Story = () => (
  <div style={{ display: 'grid', gap: 10, padding: 10, background: '#fff' }}>
    <ShimmerEffect width='10rem' />
    <ShimmerEffect width='20rem' />
    <ShimmerEffect width='5rem' />
    <ShimmerEffect width='15rem' />
    <ShimmerEffect width='25rem' />
    <ShimmerEffect width='40rem' height='20rem' />
  </div>
)
