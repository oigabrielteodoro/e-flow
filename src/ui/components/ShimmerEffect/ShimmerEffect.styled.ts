import styled, { keyframes, css } from 'styled-components'
import { darken, lighten } from 'polished'

import { ShimmerEffectProps } from '.'

type ContainerProps = Required<
  Pick<ShimmerEffectProps, 'height' | 'width' | 'color'>
>

const progress = keyframes`
  from {
      background-position: -30rem 0;
  }

  to {
      background-position: calc(30rem + 100%) 0;
  }
`

export const Container = styled.div<ContainerProps>`
  animation: ${progress} 1.2s ease-in-out infinite;
  background-image: linear-gradient(
    90deg,
    rgba(0, 0, 0, 0),
    ${({ color }) => lighten(1, color)},
    rgba(0, 0, 0, 0)
  );
  background-color: ${({ color }) => darken(0.3, color)};
  background-size: 20rem 100%;
  background-repeat: no-repeat;
  border-radius: 0.313rem;
  display: inline-block;
  line-height: 1;
  opacity: 0.1;

  ${({ height, width }) => css`
    width: ${width};
    height: ${height};
  `}
`
