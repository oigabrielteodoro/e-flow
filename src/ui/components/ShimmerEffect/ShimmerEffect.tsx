import React, { ReactNode } from 'react'

import * as S from './ShimmerEffect.styled'

export type ShimmerEffectProps = {
  isLoading?: boolean
  children?: ReactNode
  height?: string
  width?: string
  color?: string
}

export function ShimmerEffect({
  isLoading = true,
  children,
  height = '2rem',
  width = '100%',
  color = '#fff',
}: ShimmerEffectProps) {
  if (isLoading) {
    return <S.Container height={height} width={width} color={color} />
  }

  return <>{children}</>
}
