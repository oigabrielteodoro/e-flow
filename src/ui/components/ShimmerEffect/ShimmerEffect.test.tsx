import React from 'react'
import { render, screen } from '@testing-library/react'

import { ShimmerEffect } from './ShimmerEffect'

describe('<ShimmerEffect />', () => {
  it('should render correctly', () => {
    const { container } = render(<ShimmerEffect isLoading />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render when isLoading is false', () => {
    render(<ShimmerEffect isLoading={false}>Example</ShimmerEffect>)

    expect(screen.getByText(/example/i)).toBeInTheDocument()
  })
})
