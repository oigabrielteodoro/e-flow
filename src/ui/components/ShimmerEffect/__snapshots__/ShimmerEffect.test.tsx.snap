// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShimmerEffect /> should render correctly 1`] = `
.c0 {
  -webkit-animation: ePEqAU 1.2s ease-in-out infinite;
  animation: ePEqAU 1.2s ease-in-out infinite;
  background-image: linear-gradient( 90deg,rgba(0,0,0,0),#fff,rgba(0,0,0,0) );
  background-color: #b3b3b3;
  background-size: 20rem 100%;
  background-repeat: no-repeat;
  border-radius: 0.313rem;
  display: inline-block;
  line-height: 1;
  opacity: 0.1;
  width: 100%;
  height: 2rem;
}

<div
  class="c0"
  color="#fff"
  height="2rem"
  width="100%"
/>
`;
